src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/Base64Utility.cs

```diff
@@ -2,46 +2,26 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Collections.Generic;
+using System.Diagnostics;
 
 namespace System.Text.RegularExpressions.Symbolic
 {
-    /// <summary>
-    /// Custom serialization and deserialization for numbers and arrays of numbers using a base-64 alphabet similar to a decimal alphabet.
-    /// </summary>
-    internal static class Base64
+    /// <summary>Custom serialization and deserialization for numbers and arrays of numbers using a base-64 alphabet similar to a decimal alphabet.</summary>
+    internal static class Base64Utility
     {
-        private static readonly char[] s_customBase64encoding = new char[64] {
-            '0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
-            'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z',
-            'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z',
-            '+', '/'
-        };
-
-        private static ReadOnlySpan<byte> CustomBase64Decoding => new byte[128] {
-            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-            62, //'+' maps to 62
-            0, 0, 0,
-            63, //'/' maps to 63
-            0, 1, 2, 3, 4, 5, 6, 7, 8, 9, //digits map to 0..9
-            0, 0, 0, 0, 0, 0, 0,
-            10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, //uppercase letters
-            0, 0, 0, 0, 0, 0,
-            36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, //lowercase letters
-            0, 0, 0, 0, 0
-        };
-
         /// <summary>
         /// Custom Base64 encoder for int.
         /// It uses the standard base 64 alphabet but reordered to 0-9A-Za-z+/ so numbers up to 15 (0-9A-F) coincide with hexadecimal digits.
         /// Negative numbers start with '-'.
         /// </summary>
         public static void Encode(int n, StringBuilder builder)
         {
+            Debug.Assert(n != int.MinValue);
```

Comment:
@veanes, I added this assert because Encode will not work correctly in this case, but does anything prevent it from happening?

Reply:
From where Encode is used currently (essentially for small integers) the assert cannot arise (unless there is a buggy scenario I cannot see ... what the assert is for). I don't think it is even used for **any** negative numbers although the type is int -- but I'm not 100% sure here ,, but **definitely not** int.Minvalue.
(used for character codes, length and similar).